src/System.Text.Json/tests/Serialization/Stream.ReadTests.cs

```diff
@@ -139,5 +139,98 @@ public static async Task TestBOMWithNoJsonValue()
                     async () => await JsonSerializer.DeserializeAsync<byte>(stream));
             }
         }
+
+        public static IEnumerable<object[]> BOMWithJsonElementTestData
```

Comment:
nit: This test data doesn't have much to do with `JsonElement`. Consider renaming.